lib/nexpose/scheduled_backup.rb

```diff
@@ -0,0 +1,79 @@
+module Nexpose
+# Constants useful across the Nexpose module.
+# Configuration structure for scheduled backups.
+  class ScheduledBackup < APIObject
+    require 'json'
+    include JsonSerializer
+
+    # Whether or not this schedule is enabled.
+    attr_accessor :enabled
+    # Valid schedule types: daily, hourly, monthly-date, monthly-day, weekly.
+    attr_accessor :schedule_type
+    # The repeat interval based upon type.
+    attr_accessor :schedule_interval
+    # The earliest date to generate the report on (in ISO 8601 format).
+    attr_accessor :schedule_start
+    # The description of the backup
+    attr_accessor :description
+    # Whether the backup will be platform independent or not
+    attr_accessor :platform_independent
+    # Whether the backup should pause all local scans or wait for local scans to complete.
+    attr_accessor :pause_local_scans
+    # Cancellation window in minutes for scheduled DB maintenance.
+    attr_accessor :cancellation_period
+
+    def initialize(start, enabled=true, type, interval, platform_independent, description, pause_local_scans, cancellation_period)
```

Comment:
Given the number of arguments needed it would be better to use keyword arguments: https://robots.thoughtbot.com/ruby-2-keyword-arguments

Also is it possible to reduce the number of required arguments by using some defaults? I think the following could have defaults that the user can ignore setting when creating the schedule object:

* enabled=true
* platform_independent=true
* description=nil
* pause_local_scans=true
* cancellation_period=0 or nil (whatever is acceptable for "not set")